pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -762,11 +770,16 @@ type NodeStatus struct {
 	TemplateRef *TemplateRef `json:"templateRef,omitempty" protobuf:"bytes,6,opt,name=templateRef"`
 
 	// StoredTemplateID is the ID of stored template.
+	// DEPRECATED: This value is not used anymore.
 	StoredTemplateID string `json:"storedTemplateID,omitempty" protobuf:"bytes,18,opt,name=storedTemplateID"`
 
 	// WorkflowTemplateName is the WorkflowTemplate resource name on which the resolved template of this node is retrieved.
+	// DEPRECATED: This value is not used anymore.
 	WorkflowTemplateName string `json:"workflowTemplateName,omitempty" protobuf:"bytes,19,opt,name=workflowTemplateName"`
 
+	// TemplateScope is the template scope in which the template of this node was retrieved.
+	TemplateScope string `json:"templateScope,omitempty" protobuf:"bytes,20,opt,name=templateScope"`
```

Comment:
`TemplateScope` is used to resolve workflow-template-local templates. Because we use 2 types of references: `Template` and `TemplateRef`, this field is necessary. If we can remove `Template` and all the references are made with `TemplateRef`, we don't need `TemplateScope` and simplify the code. What do you think?

Reply:
I still not fully understand the need for this.
We can just have resolution logic where `.TemplateRef` is checked only when `.Template` is empty.